### ISSUE TITLE
security: password in memory

### ISSUE THREAD

**Issue by c0d3z3r0** [iksaif/thinkpad-wmi#7]
> 
> password
> Must contain the BIOS supervisor password (aka 'pap'), if set, to be able to do any change.
> 
> Every subsequent password change will be authorized with this password. The password may be unloaded by writing an empty string. Writing an invalid password may trigger the BIOS' invalid password limit, such that a reboot will be required in order to make any further BIOS changes.

From security perspective leaving the password in memory is not a good idea.
IMHO every write should contain the password / auth string.

Any other opinions?

**Comment by iksaif** [comment]
Makes sense, but this is going to make the current debugfs interface quite hard to use. If there was another interface (char device?) then clearly that would be the way to go.

**Comment by c0d3z3r0** [comment]
Ah, I'm sorry. I assumed having a char device... #7 depends on #8
Not sure if the debugfs interface should be kept at all
